app/models/event_schedule.rb

```diff
@@ -37,6 +39,16 @@ def replacement?
 
   private
 
+  def start_after_end_hour
+    return unless errors.blank? && event.program.conference.try(:end_hour)
```

Comment:
Yes we can do that but still then we would have to check the presence of `start_time` in the validation body so basically we will have if condition in validation definition as well as in validation body so I don't know if that's okay, may be we should have the if condition only in the validation body like earlier. But yes I think `errors` approach is something we should not go with. Actually I myself wasn't convinced with that approach as it requires all previous validations to pass successfully which is not desired.
So I think we should switch back to our previous approach what do you think?